src/main/java/com/gd/clinic/controller/AdminController.java

```
@@ -1,22 +1,82 @@
 package com.gd.clinic.controller;
 
 import com.gd.clinic.api.AdminApi;
-import com.gd.clinic.model.CredentialsDto;
-import com.gd.clinic.model.NewUserDto;
-import com.gd.clinic.model.UserResponseDto;
+import com.gd.clinic.model.*;
+import com.gd.clinic.security.entity.User;
+import com.gd.clinic.security.enums.RoleName;
+import com.gd.clinic.security.service.RefreshTokenService;
+import com.gd.clinic.security.service.RoleService;
+import com.gd.clinic.security.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.time.OffsetDateTime;
+
+
 @RestController
 public class AdminController implements AdminApi {
 
+    @Autowired
+    UserService userService;
+    @Autowired
+    RoleService roleService;
+
+    @Autowired
+    RefreshTokenService refreshTokenService;
+
+
     @Override
+    @PreAuthorize("hasRole('ADMIN')")
     public ResponseEntity<UserResponseDto> createUser(NewUserDto newUserDto) {
-        return null;
+        String role = "";
+        User user = configUser(newUserDto);
+        if(userService.existsByUsername(newUserDto.getUserName())){
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
```

this is responsibility if a service, and better to throw an exception